runtime/themes/onedark.toml

```diff
@@ -40,7 +40,7 @@
 "diff.delta" = "gold"
 "diff.minus" = "red"
 
-diagnostic = { modifiers = ["underlined"] }
+diagnostic = { modifiers = ["undercurled"] }
```

Comment:
What happens if the user's terminal doesn't support undercurl?

Reply:
We should look up `Smulx` in terminfo then fallback all of these to underline if not present: https://github.com/wez/wezterm/issues/415#issuecomment-755442430=

Reply:
Crossterm [doesn't support querying terminfo](https://github.com/crossterm-rs/crossterm/issues/495), so we would have to pull in something like the [`terminfo`](https://docs.rs/terminfo/latest/terminfo/index.html) crate to do this. Unless there's a better (no-dependency) way to this?

Reply:
I was thinking maybe the `term` crate since it has less dependencies (no nom or phf). We could also inline a crate, it would be useful for detecting other features more reliably (true color)

Reply:
The `term` crate cannot parse extended capabilities like `Smulx`, so we would have to contribute it upstream first. Another option is to use the [`cxterminfo`][1] crate which has support for [extended capabilities][2] and is quite well documented too. Another advantage of `cxterminfo` is that it's only about 500 lines long and has no dependencies so it would be easy to inline into helix.

[1]: https://github.com/BxNiom/cx-terminfo
[2]: https://github.com/BxNiom/cx-terminfo/blob/9fee0c06544dea9c5b5b25224132b51fdf3df548/src/terminfo.rs#L293-L297
[3]: https://github.com/BxNiom/cx-terminfo/blob/9fee0c06544dea9c5b5b25224132b51fdf3df548/src/terminfo.rs

Reply:
That seems like a good option 👍🏻